File: Python/boj/Gold/2447.py
import sys

if __name__ == "__main__":
    try:
        sys.stdin = open("input.txt", "r")
    except FileNotFoundError:
        pass

    N = int(input())
    result = []

    def star(row, col, size):
        if row % 3 == 1 and col % 3 == 1:
            result.append(" ")
        elif size == 1:
            result.append("*")
        else:
            star(row // 3, col // 3, size // 3)

    for i in range(N):
        for j in range(N):
            star(i, j, N)
        result.append("\n")

    print("".join(result))

File: JavaScript/boj/Gold/2447.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const N = fs.readFileSync(filePath).toString().trim().split('\n');

const result = [];

const star = (row, col, size) => {
  if (row % 3 === 1 && col % 3 === 1) result.push(' ');
  else if (size === 1) result.push('*');
  else star(parseInt(row / 3), parseInt(col / 3), size / 3);
};

for (let i = 0; i < N; i++) {
  for (let j = 0; j < N; j++) star(i, j, N);
  result.push('\n');
}

console.log(result.join(''));
